lib/data/services/registration_service_impl.dart

```diff
@@ -0,0 +1,14 @@
+import 'package:firebase_auth/firebase_auth.dart';
+import 'package:water_tracker/data/services/registration_service.dart';
+
+class RegistrationServiceFirebaseImpl extends RegistrationService {
+  final credential = FirebaseAuth.instance;
```

Comment:
not sure about credential name. Mby auth?

Reply:
I like my option more